src/components/ui/date-field.tsx

```
@@ -0,0 +1,149 @@
+  const [month, setMonth] = useState(
+    date ? (date.getMonth() + 1).toString().padStart(2, "0") : "",
+  );
+  const [year, setYear] = useState(date ? date.getFullYear().toString() : "");
+
+  const isValidDate = (year: string, month: string, day: string): boolean => {
+    const parsedYear = parseInt(year, 10);
+    const parsedMonth = parseInt(month, 10) - 1;
+    const parsedDay = parseInt(day, 10);
+    const testDate = new Date(parsedYear, parsedMonth, parsedDay);
+    return (
+      !isNaN(testDate.getTime()) &&
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)